File: src/main/java/project/rpg/commands/StatusCommand.java
package project.rpg.commands;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import project.rpg.manager.AttributeManager;
import project.rpg.player.info.Status;

import java.util.UUID;

public class StatusCommand implements CommandExecutor {

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player) {
            Player player = (Player) sender;
            UUID uuid = player.getUniqueId();

            if ("add".equals(args[0])) {
                int num = Integer.parseInt(args[2]);
                switch (args[1]) {
                    case "strength":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addStrength(num);
                        break;
                    case "agility":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addAgility(num);
                        break;
                    case "speed":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addSpeed(num);
                        break;
                    case "health":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addHealth(num);
                        break;
                    case "defense":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addDefense(num);
                        break;
                    case "luck":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addLuck(num);
                        break;
                    case "handicraft":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addHandicraft(num);
                        break;
                    case "attractive":
                        player.sendMessage(args[1] + " is added " + args[2]);
                        Status.getPlayerMap().get(uuid).addAttractive(num);
                        break;
                    default:
                        sender.sendMessage("오타난 커멘드");
                        break;
                }
                //jsonFile_.put(playerName, playerData_.get(playerName).getMap());
                AttributeManager.setAttributes(player, Status.getPlayerMap().get(uuid));
            }
            return true;
        }
        return false;
    }
}

File: src/main/kotlin/project/rpg/commands/StatusCommand.kt
package project.rpg.commands

import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.entity.Player
import project.rpg.player.status

class StatusCommand : CommandExecutor, TabCompleter {

    private lateinit var arg1: String
    private lateinit var arg2: String
    private lateinit var player: Player

    private val recommendations = listOf(
        listOf("add"),
        listOf("strength", "agility", "speed", "health", "defense", "luck", "handicraft", "intelligence")
    )

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {

        if (sender is Player) {
            player = sender
            val status = player.status

            if ("add" == args[0]) {
                if (args.size <= 2) {
                    sender.sendMessage(command.usage)
                    return true
                }
                this.arg1 = args[1]
                this.arg2 = args[2]

                val num = arg2.toInt()
                if (status.addStatus(arg1, num)) sendFeedback(true)
                status.applyChanges()
            } else sendFeedback(false)
            return true
        }
        return false
    }

    private fun sendFeedback(isSuccess: Boolean) {
        if (isSuccess) player.sendMessage("$arg1 is added $arg2")
        else player.sendMessage("오타난 커맨드")
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>?
    ): List<String> {
        return if (args != null) {
            when (args.size) {
                1 -> recommendations[0]
                2 -> recommendations[1]
                else -> ArrayList()
            }
        }
        else ArrayList()
    }
}